src/NServiceBus.Core/Transports/Receiving.cs

```diff
@@ -15,17 +15,24 @@ internal Receiving()
             Defaults(s =>
             {
                 var transportInfrastructure = s.Get<TransportInfrastructure>();
-                var userDiscriminator = s.GetOrDefault<string>("EndpointInstanceDiscriminator");
-                var baseQueueName = s.GetOrDefault<string>("BaseInputQueueName") ?? s.EndpointName();
-                if (userDiscriminator != null)
+                var discriminator = s.GetOrDefault<string>("EndpointInstanceDiscriminator");
+                var baseQueueName = s.GetOrDefault<string>("BaseInputQueueName");
+                var mainInstance = new EndpointInstance(s.EndpointName());
+                if (baseQueueName != null)
                 {
-                    var p = s.Get<TransportInfrastructure>().BindToLocalEndpoint(new EndpointInstance(baseQueueName, userDiscriminator));
-                    s.SetDefault("NServiceBus.EndpointSpecificQueue", transportInfrastructure.ToTransportAddress(new LogicalAddress(p)));
+                    mainInstance = mainInstance.SetProperty("queue", baseQueueName);
```

Comment:
I don't think the Properties are a good way to expose core concepts on `EndpointInstance`. In that case it should be a property on EndpointInstance directly as you have to ensure that all transports know about this property when implementing `ToTransportAddress`.

Can't we keep routing and input queue setup seperated (use "queue" property for transport specific routing, use a core only concept to implement `OverrideLocalAddress`).